content-services/content-sources-backend
# Fixes 1846: log request ID

## Summary
- Adds support for the request ID, `x-rh-insights-request-id`, sent in the header and propagates that request ID to logging in the API, tasking, and redis
- Also propagates TaskID/Type logging throughout tasking system and pulp. 

The goal here is to more easily track which events correspond to which request. For example, if I get an error during repository snapshot, it is nice to know which request or task triggered that failed snapshot.

TODO
- [x] I need to double check redis logging. Will add testing step for that as well.


## Testing steps
1. Make an API request, for example, one to create a repository. Add the `x-rh-insights-request-id` header.
```
POST http://localhost:8000/api/content-sources/v1.0/repositories/
x-rh-identity: zEzcvAe....
x-Rh-Insights-Request-Id: 12345
Content-Type: application/json
```
3. You should see the request ID propagated through the to the logs for the API, tasking, and introspection. Set your log level to  `debug` to make sure you see all the logs.
4. Try variations of snapshotting, introspecting, CTRL+C exiting the server while it's in the middle of a task, and make sure the logs are all showing the request ID (or task ID if applicable).
5. Provide feedback if you think something should or shouldn't be logged.
6. A quick test for RBAC/Redis logging is to hard-code an error in `client_wrapper.go` and see the error get logged.
